odl/solvers/util/callback.py

```diff
@@ -440,13 +440,20 @@ def __init__(self, *args, **kwargs):
             where ``cur_iter_num`` is the current iteration number.
         step : positive int, optional
             Number of iterations between plots. Default: 1
-        saveto : string, optional
+        saveto : string or callable, optional
             Format string for the name of the file(s) where
-            iterates are saved. The file name is generated as
+            iterates are saved.
+
+            If saveto is a string, the file name is generated as::
 
                 filename = saveto.format(cur_iter_num)
 
             where ``cur_iter_num`` is the current iteration number.
+
+            If saveto is a callable, the file name is generated as::
+
+                filename = saveto(cur_iter_num)
+
             If the directory name does not exist, a ``ValueError`` is raised.
```

Comment:
The note is helpful, yes. But we do whatever matplotlib does, we shouldn't be too specific here I think. Perhaps change it to something like "Note that an error will be raised when trying to create a new directory."